crates/biome_js_analyze/src/lint/nursery/use_adjacent_getter_setter.rs

```diff
@@ -0,0 +1,490 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_js_syntax::static_value::StaticValue;
+use biome_js_syntax::{
+    AnyJsClassMember, AnyJsClassMemberName, AnyJsObjectMember, AnyJsObjectMemberName,
+    AnyTsTypeMember, JsClassMemberList, JsComputedMemberName, JsGetterClassMember,
+    JsGetterObjectMember, JsLiteralMemberName, JsMetavariable, JsObjectMemberList,
+    JsPrivateClassMemberName, JsSetterClassMember, JsSetterObjectMember,
+    TsGetterSignatureClassMember, TsGetterSignatureTypeMember, TsSetterSignatureClassMember,
+    TsSetterSignatureTypeMember, TsTypeMemberList,
+};
+use biome_rowan::{AstNode, AstNodeList, AstSeparatedList, SyntaxResult, declare_node_union};
+
+declare_lint_rule! {
+    /// Enforce that getters and setters for the same property are adjacent in class and object definitions.
+    ///
+    /// When defining a property in a class or object, it's common to have both a getter and a setter.
+    /// This rule enforces that getter is defined right before the setter,
+    /// making the code more maintainable and easier to read.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// Name getter and setter are not adjacent:
+    ///
+    /// ```js,expect_diagnostic
+    /// class User {
+    ///   get name() { return this._name; }
+    ///   constructor() {}
+    ///   set name(value) { this._name = value; }
+    /// }
+    /// ```
+    ///
+    /// Getter should go before the setter.
+    ///
+    /// ```js,expect_diagnostic
+    /// const user = {
+    ///   set name(value) { this._name = value; },
+    ///   get name() { return this._name; }
+    /// };
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// class User {
+    ///   get name() { return this._name; }
+    ///   set name(value) { this._name = value; }
+    ///   get age() { return this._age; }
+    ///   set age(age) { this._age = age; }
+    /// }
+    /// ```
+    ///
+    /// This rule does not enforce the existence of both getter and setter for a property.
+    /// Single getters without setters and setters without getters are ignored.
+    ///
+    pub UseAdjacentGetterSetter {
+        version: "next",
```

Comment:
This should have been changed to 2.0.0. Can you please update it?

Reply:
Sure, here is it: https://github.com/biomejs/biome/pull/6046

What's the rule there? When should I specify `next` and when `2.0.0`?

Reply:
Usually you specify always next, then right before a release we bump all of them to the next version number.

However, this is an unusual situation where we bumped the rules and still haven't made a production release.

It's not a big deal, it's just a metadata that affects the documentation